### PR TITLE
Add support for torrent-rename-path rpc

### PR DESCRIPTION
**Pull request by JohanAR**
The Transmission RPC API only supports changing the name of files and
directories, moving is not possible.

Solves #2

### CODE REVIEW

**Review by ajf8 — APPROVED** [review]

File: src/trg-files-model.c [hunk 1]
```diff
@@ -83,6 +83,30 @@ static void trg_files_update_parent_progress(GtkTreeModel * model,
...
+
+    while (1) {
+        GtkTreeIter tmp_iter;
+
+        if (!gtk_tree_model_iter_parent(model, &tmp_iter, &back_iter))
+            break;
+
+        --path_last;
+
+        /* Is it better to test if name has changed before calling store_set? */
```
Comment (ajf8) [inline src/trg-files-model.c:102]: Should be fine, the models don't change under you in the gtk main loop iirc

**Comment by ajf8** [comment]
Thanks for this Johan, nice to see a new feature :) will merge it soon and perhaps do a release

**Comment by ajf8** [comment]
I tried it out and it seems to work well

**Comment by JohanAR** [comment]
Thanks! I hope I didn't make any mistakes, I haven't programmed C in years and never worked with GTK before :)

**Comment by joelchrono12** [comment]
I am trying to rename a folder and I don't get anything by doing right click or hitting Function 2. How do I use this? sorry if its explained somewhere

**Pull request merged**